File: C++/Data Structure/Bubble Sort.cpp
#include<iostream>
using namespace std;
void swapping(int &a, int &b) {      
   int temp;
   temp = a;
   a = b;
   b = temp;
}
void display(int *array, int size) {
   for(int i = 0; i<size; i++)
      cout << array[i] << " ";
   cout << endl;
}
void bubbleSort(int *array, int size) {
   for(int i = 0; i<size; i++) {
      int swaps = 0;         
      for(int j = 0; j<size-i-1; j++) {
         if(array[j] > array[j+1]) {      
            swapping(array[j], array[j+1]);
            swaps = 1;   
         }
      }
      if(!swaps)
         break;    
   }
}
int main() {
   int n;
   cout << "Enter the number of elements: ";
   cin >> n;
   int arr[n];     
   cout << "Enter elements:" << endl;
   for(int i = 0; i<n; i++) {
      cin >> arr[i];
   }
   cout << "Array before Sorting: ";
   display(arr, n);
   bubbleSort(arr, n);
   cout << "Array after Sorting: ";
   display(arr, n);
}

File: Rust/Algorithms/Sorting/bubble_sort.rs
fn main() {
    let mut v = Vec::new();
    v.push(3);
    v.push(7);
    v.push(-1);
    v.push(50);
    v.push(1);
    v.push(8);
    bubble_sort(&mut v);
    println!("{:?}", v);
}

fn bubble_sort(v: &mut Vec<i32>) {
    for i in 0..v.len() {
        for j in i..v.len() {
            if v[i] > v[j] {
                let temp = v[i];
                v[i] = v[j];
                v[j] = temp;
            }
        }
    }
}
